src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyLineWrappingInBlockCheck.java

```diff
@@ -0,0 +1,323 @@
+                      || line.trim().charAt(0) == '*'))
+                      || line.trim().startsWith(INLINE_COMMENT_START)) {
+                    index--;
+                    line = fileContents.getLine(index);
+                }
+                if(line.trim().startsWith(MULTILINE_COMMENT_START)) {
+                    index--;
+                    line = fileContents.getLine(index);
+                }
+            }
+            result = CommonUtil.isBlank(line);
+        }
+        return result;
+    }
+
+    /**
+     * Checks whether an empty/blank line is present after the given block.
+     *
+     * @param ast the current node
+     * @return true if an empty/blank line is present after the given block.
+     */
+    private boolean hasEmptyLineAfter(DetailAST ast) {
+        boolean result = false;
+        final DetailAST rightCurly = getRightCurly(ast);
+        final FileContents fileContents = getFileContents();
+        final int index = rightCurly.getLineNo();
+        if (index < fileContents.getLines().length) {
+            final String text = fileContents.getLine(index);
+            if (text.length() == 0) {
+                result = true;
+            }
+            else {
+                result = fileContents.lineIsBlank(index);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Finds the respective right curly AST node of the given block.
+     *
+     * @param detailAST the current node
+     * @return the right curly AST node of the given block.
+     */
+    private DetailAST getRightCurly(DetailAST detailAST) {
+        DetailAST rCurly = null;
+        // As the right curly is not present in the same position for all blocks
+        switch (detailAST.getType()) {
+            case TokenTypes.LITERAL_TRY:
+            case TokenTypes.LITERAL_IF:
+            case TokenTypes.LITERAL_ELSE:
+            case TokenTypes.LITERAL_SWITCH:
+            case TokenTypes.METHOD_DEF:
+            case TokenTypes.CTOR_DEF:
+            case TokenTypes.LITERAL_FOR:
+            case TokenTypes.LITERAL_WHILE:
+            case TokenTypes.LITERAL_DO:
+            case TokenTypes.STATIC_INIT:
+            case TokenTypes.INSTANCE_INIT:
+                final DetailAST sList = detailAST.findFirstToken(TokenTypes.SLIST);
+                rCurly = sList.getLastChild();
+                break;
+
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.INTERFACE_DEF:
+            case TokenTypes.ENUM_DEF:
+            case TokenTypes.ANNOTATION_DEF:
+                final DetailAST objBlock = detailAST.findFirstToken(TokenTypes.OBJBLOCK);
+                rCurly = objBlock.getLastChild();
+                break;
+
+            default:
+                final String exceptionMsg = "Unexpected token type: " + detailAST.getText();
+                throw new IllegalArgumentException(exceptionMsg);
+        }
+
+        return rCurly;
+    }
```

Comment:
You also need the left brace:
```
    /**
     * Finds the left curly brace of a code block associated with the specified token.
     *
     * @param ast the token to process
     * @return the token for the left curly brace if found, {@code null} otherwise
     */
    private static DetailAST findLeftCurly(DetailAST ast) {
        DetailAST leftBrace = null;
        switch (ast.getType()) {
        case TokenTypes.ANNOTATION_ARRAY_INIT:
        case TokenTypes.ARRAY_INIT:
            leftBrace = ast;
            break;

        case TokenTypes.INTERFACE_DEF:
        case TokenTypes.CLASS_DEF:
        case TokenTypes.ANNOTATION_DEF:
        case TokenTypes.ENUM_DEF:
        case TokenTypes.ENUM_CONSTANT_DEF:
        case TokenTypes.LITERAL_NEW:
            final DetailAST objBlock = ast.findFirstToken(TokenTypes.OBJBLOCK);
            if (objBlock != null) {
                leftBrace = objBlock.getFirstChild();
            }
            break;

        case TokenTypes.LITERAL_ELSE:
            // Special handling for the "else if" case.
            final DetailAST candidate = ast.getFirstChild();
            if (candidate.getType() == TokenTypes.SLIST) {
                leftBrace = candidate;
            }
            break;

        case TokenTypes.SLIST:
            // The scope block is a SLIST with another SLIST as the parent.
            if (ast.getParent().getType() == TokenTypes.SLIST) {
                leftBrace = ast;
            }
            break;

        case TokenTypes.LITERAL_SWITCH:
            // The Switch does not have SLIST unlike the others.
            leftBrace = ast.findFirstToken(TokenTypes.LCURLY);
            break;

        case TokenTypes.CASE_GROUP:
            // The CASE_GROUP is the parent of 'case', 'default' and SLIST
            // parent.
            leftBrace = ast.getLastChild().findFirstToken(TokenTypes.SLIST);
            break;

        case TokenTypes.LITERAL_CASE:
        case TokenTypes.LITERAL_DEFAULT:
            // The 'case' and 'default' are siblings for SLIST parent.
            leftBrace = ast.getParent().getLastChild().findFirstToken(TokenTypes.SLIST);
            break;

        default:
            // All the rest: for/while/if/STATIC_INIT,lambda,etc.
            leftBrace = ast.findFirstToken(TokenTypes.SLIST);
            break;
        }

        return leftBrace;
    }
```

Reply:
Why do we need to find `left curly`?
We can check the previous line with the help of current block node.
If we check the previous line with `left curly`, we'll have to manage various cases, like if the `left curly` is present on the next line of the method then we will have to find the start of the block. This leads us to the same situation, checking the line before the current node.
Hence I thought checking with the help of current node would be much easier and better.
What do you think ?

Reply:
@pbludov Got it, why we need to find `left curly`. I misunderstood. 
Thanks for the guidance 👍